src/Migrations/M028MovePageHeaderSideBarOptions.php

```diff
@@ -0,0 +1,236 @@
+
+                ' . (!empty($page_header_description) ?
+                    '<!-- wp:paragraph {"placeholder":"Enter description","style":{"typography":{"fontSize":"1.25rem"}}} -->
+                        <p style="font-size:1.25rem">
+                            ' . $page_header_description . '
+                        </p>
+                    <!-- /wp:paragraph -->'
+                : '') . '
+
+                ' . (!empty($page_header_button_link) ?
+                    '<!-- wp:buttons -->
+                        <div class="wp-block-buttons">
+                            <!-- wp:button {"className":"is-style-cta"} /-->
+                        </div>
+                        <div class="wp-block-button is-style-cta">
+                            <a class="wp-block-button__link wp-element-button"
+                                href="' . $page_header_button_link . '"
+                                target="' . ($page_header_button_link_checkbox === 'on' ? '_blank' : '') . '"
+                                rel="noreferrer noopener"
+                            >
+                                ' . $page_header_button_title . '
+                            </a>
+                        </div>
+                    <!-- /wp:buttons -->'
+                : '') . '
+
+                </div>
+                <figure class="wp-block-media-text__media">
+                    <img
+                        src="' . $page_header_image . '"
+                        alt=""
+                        class="wp-image-' . $page_header_image_id . ' size-full"
+                    />
+                </figure>
+                </div>
+                <!-- /wp:media-text -->
+                </div>
+                <!-- /wp:group -->
+                </div>
+                <!-- /wp:group -->
+                <!-- /wp:planet4-block-templates/page-header -->
+                ';
+
+                $updated_post_content = $content . $page->post_content;
+
+                self::update_meta_options($page_meta_data, $page_header_title, $page, true);
+
+                $post_args = array(
+                    'post_content' => $updated_post_content,
+                    'ID' => $page->ID,
+                );
+
+                wp_update_post($post_args);
+
+                continue;
+            }
+
+            if (empty($page_header_image_id) || empty($page_header_title)) {
+                if (
+                    empty($page_header_image_id)
+                    && empty($page_header_title)
+                    && empty($page_header_description)
+                    && empty($page_header_button_title)
+                ) {
+                    continue;
+                }
+
+                $content = '
+                    <!-- wp:group {"layout":{"type":"flex","orientation":"vertical"}} -->
+                    <div class="wp-block-group">
+                    ' . (!empty($page_header_description) ?
+                            '<!-- wp:paragraph -->
```

Comment:
On the test instance (haven't test this locally yet) I see the whole group block being "under recovery" in the editor (eg. in the "Terms" page). When trying to recover it fails and the content is lost.

![Screenshot from 2024-03-06 15-13-40](https://github.com/greenpeace/planet4-master-theme/assets/939357/ed0c0447-193e-4974-a73f-6fcac71ef00d)


Maybe it's because of the double `<p>` markup? I think if you just have `<!-- wp:paragraph -->` you may not need to add a `<p>` as well. I tried re-creating the group block with a paragraph block. This how the two markups look on the code editor (the one on the top is the one from this migration that is broken in the visual editor).

![Screenshot from 2024-03-06 15-13-40](https://github.com/greenpeace/planet4-master-theme/assets/939357/83ac7b8f-9168-4d98-8261-4e0eb37ca60a)

Reply:
Oh I see, but what if the description does not have the `<p>`,  will that still render correctly?

Maybe we could strip the outer `p` tag in the description if it has one so that way there are no duplicates.

Reply:
Ah yes, that sounds like a much safer option if that extra `<p>` is coming from the description field. The field is not WYSIWYG so we could strip html tags in general.